lib/apollo-federation/schema.rb

```diff
@@ -53,20 +55,53 @@ def query(new_query_object = nil)
         federation_query_object
       end
 
+      def compose_directives
+        @compose_directives || find_inherited_value(:compose_directives, [])
+      end
+
+      def links
+        @links || find_inherited_value(:links, [])
+      end
+
+      def all_links
+        imported_directives = IMPORTED_DIRECTIVES
+        default_link = {
+          url: "https://specs.apollo.dev/federation/v#{federation_version}",
+          import: imported_directives,
+        }
+        default_link[:as] = default_link_namespace if default_link_namespace != DEFAULT_LINK_NAMESPACE
+        [default_link, *links]
+      end
+
       private
 
       def original_query
         @orig_query_object || find_inherited_value(:original_query)
       end
 
       def federation_2_prefix
-        federation_namespace = ", as: \"#{link_namespace}\"" if link_namespace != DEFAULT_LINK_NAMESPACE
+        schema = ['extend schema']
+
+        all_links.each do |link|
+          link_str = '  @link('
+          link_str += "url: \"#{link[:url]}\""
+          link_str += ", as: \"#{link[:as]}\"" if link[:as]
+          if link[:import]
+            imported_directives = link[:import].map { |d| "\"@#{d}\"" }.join(', ')
+            link_str += ", import: [#{imported_directives}]"
+          end
+          link_str += ')'
+          schema << link_str
+        end
+
+        compose_directives.each do |directive|
+          schema << "  @composeDirective(name: \"@#{directive}\")"
+        end
 
-        <<~SCHEMA
-          extend schema
-            @link(url: "https://specs.apollo.dev/federation/v#{federation_version}"#{federation_namespace}, import: [#{(IMPORTED_DIRECTIVES.map { |directive| "\"@#{directive}\"" }).join(', ')}])
+        schema << ''
+        schema << ''
```

Comment:
why twice?

Reply:
It's replaced into two newlines to make one line break between `@composeDirective` lines and the rest of the schema